pkg/chain/chain.go

```diff
@@ -1,164 +1,239 @@
+// managing, monitoring, and creating bonded ECDSA keeps as well as the
+// operator's eligibility for participation in such keeps.
+type BondedECDSAKeepManager interface {
+	// TBTCApplicationHandle returns a handle for interacting with the tBTC
+	// application associated with this BondedECDSAKeepManager. Returns nil with
+	// an error if no tBTC application exists for this manager.
+	TBTCApplicationHandle() (TBTCHandle, error)
 
 	// OnBondedECDSAKeepCreated installs a callback that is invoked when an
 	// on-chain notification of a new bonded ECDSA keep creation is seen.
 	OnBondedECDSAKeepCreated(
 		handler func(event *BondedECDSAKeepCreatedEvent),
 	) subscription.EventSubscription
 
+	// GetKeepCount returns number of keeps.
+	GetKeepCount() (*big.Int, error)
+
+	// GetKeepAtIndex returns a handle to the keep at the given index.
+	GetKeepAtIndex(keepIndex *big.Int) (BondedECDSAKeepHandle, error)
+	// GetKeepAtIndex returns a handle to the keep with the given identifier.
+	GetKeepWithID(keepID KeepID) (BondedECDSAKeepHandle, error)
+
+	// IsOperatorAuthorized checks if the factory has the authorization to
+	// operate on stake represented by the provided operator.
+	IsOperatorAuthorized(operator OperatorID) (bool, error)
+}
+
+type BondedECDSAKeepApplicationHandle interface {
+	// ID should return the id of this application in a host chain-agnostic
+	// format.
+	ID() KeepApplicationID
+
+	// RegisterAsMemberCandidate registers client as a candidate to be selected
+	// to a keep.
+	RegisterAsMemberCandidate() error
+
 	// IsRegisteredForApplication checks if the operator is registered
 	// as a signer candidate in the factory for the given application.
-	IsRegisteredForApplication(application common.Address) (bool, error)
+	IsRegisteredForApplication() (bool, error)
 
 	// IsEligibleForApplication checks if the operator is eligible to register
 	// as a signer candidate for the given application.
-	IsEligibleForApplication(application common.Address) (bool, error)
+	IsEligibleForApplication() (bool, error)
 
 	// IsStatusUpToDateForApplication checks if the operator's status
 	// is up to date in the signers' pool of the given application.
-	IsStatusUpToDateForApplication(application common.Address) (bool, error)
+	IsStatusUpToDateForApplication() (bool, error)
 
 	// UpdateStatusForApplication updates the operator's status in the signers'
 	// pool for the given application.
-	UpdateStatusForApplication(application common.Address) error
-
-	// IsOperatorAuthorized checks if the factory has the authorization to
-	// operate on stake represented by the provided operator.
-	IsOperatorAuthorized(operator common.Address) (bool, error)
-
-	// GetKeepCount returns number of keeps.
-	GetKeepCount() (*big.Int, error)
-
-	// GetKeepAtIndex returns the address of the keep at the given index.
-	GetKeepAtIndex(keepIndex *big.Int) (common.Address, error)
+	UpdateStatusForApplication() error
```

Comment:
I'm wondering if it would be worth updating the functions names as they are exposed on the application handle now.
`IsRegisteredForApplication` -> `IsOperatorRegistered` or `IsRegistered`
`IsEligibleForApplication` -> `IsOperatorEligible` or `IsEligible`
`IsStatusUpToDateForApplication` -> `IsOperatorStatusUpToDate` or `IsStatusUpToDate`
`UpdateStatusForApplication` -> `UpdateOperatorStatus` or `UpdateStatus`

Reply:
Love it.